static/js/components/utils/AutoCompleteSettings.js

```diff
@@ -0,0 +1,125 @@
+      if (i && j) {
+        if (searchText.charAt(j - 1) === key.charAt(i - 1)) value = prev;
+        else value = Math.min(current[j], current[j - 1], prev) + 1;
+      } else {
+        value = i + j;
+      }
+      prev = current[j];
+      current[j] = value;
+    }
+  }
+  return current.pop();
+};
+AutoCompleteSettings.levenshteinDistanceFilter = (distanceLessThan) => {
+  if (distanceLessThan === undefined) {
+    return levenshteinDistance;
+  } else if (typeof distanceLessThan !== 'number') {
+    throw 'Error: AutoCompleteSettings.levenshteinDistanceFilter is a filter generator, not a filter!';
+  }
+
+  return (s, k) => levenshteinDistance(s, k) < distanceLessThan;
+};
+
+AutoCompleteSettings.fuzzyFilter = (searchText, key) => {
+  const compareString = key.toLowerCase();
+  searchText = searchText.toLowerCase();
+
+  let searchTextIndex = 0;
+  for (let index = 0; index < key.length; index++) {
+    if (compareString[index] === searchText[searchTextIndex]) {
+      searchTextIndex += 1;
+    }
+  }
+
+  return searchTextIndex === searchText.length;
+};
+
+AutoCompleteSettings.defaultMenuItemRender = (settings, option) => { // eslint-disable-line react/display-name
+  return <MenuItem
+    key={option.value}
+    primaryText={option.label}
+    value={option.value}
+    {...settings.props}
+  />;
+};
+
+const renderHighlightedOption = (optionLabel, searchText) => {
+  let searchTextIndex = optionLabel.toLowerCase().indexOf(searchText.toLowerCase());
+  if (searchTextIndex > -1) {
+    let matchStartIndex = searchTextIndex,
+      matchEndIndex = matchStartIndex + searchText.length;
+    let labelStart = optionLabel.substring(0, matchStartIndex),
+      labelMatch = optionLabel.substring(matchStartIndex, matchEndIndex),
+      labelEnd = optionLabel.substring(matchEndIndex);
+    return <span>{labelStart}<strong>{labelMatch}</strong>{labelEnd}</span>;
+  }
+  else {
+    return <span>{optionLabel}</span>;
+  }
+};
+AutoCompleteSettings.highlightedMenuItemRender = (settings, option) => { // eslint-disable-line react/display-name
+  return <MenuItem
+    key={option.value}
+    value={option.value}
+    {...settings.props}
+  >
+    { renderHighlightedOption(option.label, settings.searchText) }
+  </MenuItem>;
+};
+
+AutoCompleteSettings.showAllOptions = (props) => {
+  const { filter } = props;
+  return {
+    filter: filter || AutoCompleteSettings.caseInsensitivePrefixFilter,
+    openOnFocus: true
+  };
+};
+
+AutoCompleteSettings.showLimitedOptions = (props) => {
+  const { filter, maxSearchResults } = props;
+  return {
+    filter: filter || AutoCompleteSettings.nonBlankCaseInsPrefixFilter,
+    openOnFocus: false,
+    showOptionsWhenBlank: false,
+    maxSearchResults: maxSearchResults || DEFAULT_OPTION_LIMIT_COUNT
+  };
+};
+
+AutoCompleteSettings.highlightMatchedOptionText = () => {
+  return {
+    menuItemRenderer: AutoCompleteSettings.highlightedMenuItemRender
+  };
+};
+
+export default AutoCompleteSettings;
```

Comment:
I feel like we should `export` each of the functions above instead of exporting this object here. We will never use this object except to retrieve its functions